src/plugins/release-merge-purifier/index.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * @fileoverview Adds fail status to all non-semver pr's after release
+ * @author Gyandeep Singh
+ */
+
+"use strict";
+
+const semver = require("semver");
+
+const TAG_REGEX = /^(?:Build|Chore|Docs|Fix):/;
+
+/**
+ * Apply different checks on the commit message to see if its ok for a patch release
+ * @param {string} message - commit message
+ * @returns {boolean} `true` if the commit message is valid for patch release
+ * @private
+ */
+function isMessageValidForPatchRelease(message) {
+    return TAG_REGEX.test(message);
+}
+
+/**
+ * Get the correct commit message for a PR
+ * @param {Object} allCommits - Collection of all commit objects
+ * @param {Object} context - context object
+ * @returns {string} Commit message
+ * @private
+ */
+function getCommitMessageForPR(allCommits, context) {
+    return allCommits.data.length === 1
+        ? allCommits.data[0].commit.message
+        : context.payload.pull_request.title;
+}
+
+/**
+ * Check if the version is atleast minor
+ * @param {string} version - version to check
+ * @returns {boolean} true if its minor version
+ * @private
+ */
+function atLeastMinorVersion(version) {
+    return semver.patch(semver.clean(version)) === 0;
```

Comment:
I don't think it's correct to check whether we're in "release mode" by looking at the patch number of the latest release. For example, if we don't have any semver-minor commits for a release cycle, we would end up publishing a patch release on Friday, and then waiting until Monday to see if another patch release is necessary. In that case, the patch number would not be 0.

I think it would be better to have the release team add a label manually to enable patch-release mode. (The team already leaves a comment on the release issue saying the same thing, so adding a label doesn't seem like too much of an extra burden.)

Reply:
Good point. 
Maybe I can stop listening for published event and start listening for `issue.labeled` event and check if the label got added to release issue then do operations what i am doing right now. This way the release owner can put that label after the release is complete.
Or we can automate by adding `post-release` label whenever a release is published. Owners have to go manually remove the label or close the release issue.

Does this sound good?

Reply:
> Maybe I can stop listening for published event and start listening for `issue.labeled` event and check if the label got added to release issue then do operations what i am doing right now. This way the release owner can put that label after the release is complete.

I think this would be the best plan.